File: springapp/src/main/java/com/examly/springapp/service/OrderService.java
package com.examly.springapp.service;

import java.util.List;
import java.util.Optional;

import com.examly.springapp.model.Order;

public interface OrderService {


    public Order addOrder(Order order); // Adds a new order
    public Optional<Order> getOrderById(Long orderId); // Retrieves an order by ID
    public List<Order> getAllOrders(); // Retrieves all orders
    public Order updateOrder(Long orderId, Order updatedOrder); // Updates order details
    List<Order> getOrdersByUserId(Long userId); // Retrieves orders by user ID
    public boolean deleteOrder(Long orderId); // Deletes an order by ID

}

File: angularapp/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import { Order } from '../models/order.model';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';
import { apiUrl } from 'src/url';

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  baseUrl=apiUrl;

  constructor(private httpClient: HttpClient) { }

  placeOrder(order:Order): Observable<any>{
    return this.httpClient.post(this.baseUrl+"/api/orders",order);
  }

  getOrderById(orderId: number):Observable<any>{
    return this.httpClient.get(this.baseUrl+"/api/orders/"+orderId);
  }

  getAllOrders():Observable<any>{
    return this.httpClient.get(this.baseUrl+"/api/orders");
  }
  
  getOrdersByUserId(userId : number):Observable<any>{
    return this.httpClient.get(this.baseUrl+"/api/orders/user/"+userId);
  }

  updateOrder(orderId:number, order:Order):Observable<any>{
    return this.httpClient.put(this.baseUrl+"/api/orders/"+orderId,order);
  }

  deleteOrder(orderId: number):Observable<any>{
    return this.httpClient.delete(this.baseUrl+"/api/orders/"+orderId);
  }

  
  
  
}
